doxygen/DoxygenLayoutCpp.xml

```diff
@@ -0,0 +1,295 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<doxygenlayout version="1.0">
+  <!-- Generated by doxygen 1.9.8 -->
+  <!-- Navigation index tabs for HTML output -->
+  <navindex>
+    <tab type="user" url="BASE_URL/main/html/index.html" title="About"/>
+    <tab type="usergroup" url="BASE_URL/main/html/InstallationGuide.html" title="Getting Started">
+      <tab type="user" url="BASE_URL/main/html/MeshLibPythonSetupGuide.html" title="Python"/>
+      <tab type="user" url="BASE_URL/main/html/MeshLibSetupGuide.html" title="C++"/>
+      <tab type="user" url="BASE_URL/main/html/MeshLibCSharpSetupGuide.html" title="C#"/>
+    </tab>
+    <tab type="user" url="BASE_URL/main/html/License.html" title="License" />
+    <tab type="usergroup" url="BASE_URL/main/html/Tutorials.html" title="Tutorials">
+      <tab type="user" url="BASE_URL/main/html/FirstPluginTutorial.html" title="Create First Plugin using C++"/>
+    </tab>
+    <tab type="usergroup" url="BASE_URL/main/html/Examples.html" title="Examples">
+      <tab type="usergroup" url="BASE_URL/main/html/Examples.html#GeneralExamples" title="General Samples">
+        <tab type="user" url="@ref BasedApplication" title="Application Based on MeshLib"/>
+        <tab type="user" url="@ref ExampleMeshLoadSave" title="Mesh Load and Save"/>
+        <tab type="user" url="@ref ExampleMeshExport" title="Export Raw Triangles"/>
+        <tab type="user" url="@ref ExampleMeshModification" title="Mesh Modification"/>
+        <tab type="user" url="@ref ExampleMeshBoolean" title="Mesh Boolean"/>
+        <tab type="user" url="@ref ExampleMeshDecimate" title="Mesh Decimate"/>
+        <tab type="user" url="@ref ExampleMeshOffset" title="Mesh Offset"/>
+        <tab type="user" url="@ref ExampleMeshFillHole" title="Mesh Fill Hole"/>
+        <tab type="user" url="@ref ExampleMeshStitchHole" title="Mesh Stitch Holes"/>
+        <tab type="user" url="@ref ExampleMeshResolveDegenerations" title="Mesh Resolve Degenerations"/>
+        <tab type="user" url="@ref ExampleGlobalRegistration" title="Global Registration"/>
+        <tab type="user" url="@ref ExampleMeshICP" title="Mesh ICP"/>
+        <tab type="user" url="@ref ExampleMeshExtrude" title="Mesh Extrude"/>
+        <tab type="user" url="@ref ExampleFreeForm" title="Mesh Free Form Deformation"/>
+        <tab type="user" url="@ref ExampleLaplacian" title="Mesh Laplacian Deformation"/>
+        <tab type="user" url="@ref ExampleNoiseDenoise" title="Mesh Denoise"/>
```

Comment:
so we duplicate this part for each block

Reply:
need to think how to optimize

Reply:
reworked